driver-core/src/main/com/mongodb/client/model/Filters.java

```diff
@@ -86,7 +84,7 @@ public static <TItem> Bson eq(@Nullable final TItem value) {
      * @mongodb.driver.manual reference/operator/query/eq $eq
      */
     public static <TItem> Bson eq(final String fieldName, @Nullable final TItem value) {
-        return new SimpleEncodingFilter<>(fieldName, value);
+        return new OperatorFilter<>("$eq", fieldName, value);
```

Comment:
`$vectorSearch` does not accept the short form of `$eq`, which we are currently using. We have the following options:

1. Change the implementation of `Filters.eq`, which is what I did. See more on this change below.
2. Introduce a new `Filters.fullEq` (I am not sure how to name it), and document that users are not to use `Filters.eq` with `VectorSearchOptions.filter`. This is really ugly.
3. Introduce a whole new API for filters. Even if we ever do this, we will definitely not do this as part of the current task.

This change is controversial. On one hand, nobody should care whether we use a full form or not. On the other hand, for some reason we documented that we are not using the full form. This, combined with the [regex behavior](https://www.mongodb.com/docs/manual/reference/operator/query/eq/#match-a-regular-expression), allows users to abuse `Filters.eq` to dynamically use either equality, or regex matching, even though the method is clearly intended only for equality (`Filters.regex` is intended for regex matching). This makes the change breaking, but I am still willing to do it.

Reply:
Not supporting the short `$eq` form was intentional. Atlas will support in in the future, but not now.

Reply:
Can we annotate this as Beta, and remove it once Atlas supports the short form?

Reply:
Alexander Lukyanchikov said that allAtlas clusters will get the updated behavior, unless they are "pinned on an older version for some reason" (I am not even sure if this can be done by a user). So this should work.

Reply:
A ticket to remove `Filters.eqFull` https://jira.mongodb.org/browse/JAVA-5174.